adapter/pkg/messaging/connection.go

```diff
@@ -200,8 +204,9 @@ func maskURL(url string) string {
 	return url
 }
 
-type amqpFailoverURL struct {
-	url             string
+// AmqpFailoverURL defines the structure of an amqp failover url
+type AmqpFailoverURL struct {
```

Comment:
Do we need this to start with Uppercase?

Reply:
Yes, it is used in the internal package. https://github.com/wso2/product-microgateway/pull/2024/files#diff-293020966ab742c519fa456d25d8fb25f2efc4657c6a3b7c7f781c8ead1c19cfR49

Reply:
Fixed